File: src/agents/planner-agent.ts
import { generatePlan } from "../utils/llm";
import { PlanResult, RepoFile, Ticket } from "../core/types";

export class PlannerAgent {
  async run(ticket: Ticket, files: RepoFile[], isDetailed = false): Promise<PlanResult> {
    const steps = await generatePlan(ticket, files, isDetailed);
    return { steps };
  }
}

File: src/agents/planner_agent.py
"""Planner Agent — the architect sub-agent."""
from __future__ import annotations
from ..core.types import PlanResult, RepoFile, Ticket
from ..utils.llm import generate_plan

class PlannerAgent:
    async def run(self, ticket: Ticket, files: list[RepoFile], mode: str = "basic") -> PlanResult:
        return PlanResult(steps=await generate_plan(ticket, files, mode))
